packages/axios-extension/src/abap/ui5-abap-repository-service.ts

```diff
@@ -99,18 +99,21 @@ function encodeXmlValue(xmlValue: string): string {
  */
 export class Ui5AbapRepositoryService extends ODataService {
     public static readonly PATH = '/sap/opu/odata/UI5/ABAP_REPOSITORY_SRV';
+    private readonly publicUrl: string;
 
     /**
      * Extension of the base constructor to set preferred response format if not provided by caller.
      *
      * @param config optional base configuration for Axios
+     * @param publicUrl optional string representing the host exposing the ABAP repository
      */
-    public constructor(config?: AxiosRequestConfig) {
+    public constructor(config?: AxiosRequestConfig, publicUrl?: string) {
         config = config ?? {};
         config.headers = config.headers ?? {};
         // @see https://axios-http.com/docs/config_defaults
         config.headers['Accept'] = config.headers['Accept'] ?? 'application/json,application/xml,text/plain,*/*';
         super(config);
+        this.publicUrl = publicUrl ?? this.defaults.baseURL;
```

Comment:
Wait ... why do we need this here? Didn't we have the very same logic in the getter function on the provider

Reply:
So the `AbapServiceProvider` is aware of the `publicUrl` when its created as part of the `createForDestination` factory flow;
```
if (isAbapSystem(destination)) {
        provider = createInstance<AbapServiceProvider>(AbapServiceProvider, providerConfig);
        // For an ABAP destination flow, need to show the destination host URL property instead of the BAS host URL i.e. https://mydest.dest
        (provider as AbapServiceProvider).publicUrl = destination.Host;
    } else {
        provider = createInstance<ServiceProvider>(ServiceProvider, providerConfig);
    }
```
Thus, when `getUi5AbapRepository` is called from `AbapServiceProvider` the `publicUrl` is passed to the service instance which is created by the base service-provider.

Reply:
My thinking was that the snippet above is only called by `provider.createService` and it uses `provider.publicUrl` which in its getter has a similar check for undefined. However, assuming this constructor could also be called from somewhere else, you are right, the check needs to be there.

Minor thing: in the getter you use `||`, here you use `??`. I always forget which minor difference the operators have, therefore, please check which one makes more sense here, and then use the same in both cases. Or was it a conscious decision?